Bottom: f7665070eee7dad682c00e546bb19fb28242e431
Top:    3ef61f8fb4c34bad16524fd56bd5a0541c917bb8
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 20:35:09 +0000

hid/gtk: Disable GL alpha-blending for the pinout preview

We aren't using subcompositing for this, so don't alpha-blend.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 409dcac..be8f39e 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1083,8 +1083,15 @@ ghid_pinout_preview_expose (GtkWidget *widget,
   ghid_start_drawing (gport, widget);
   hidgl_start_render ();
 
+#if 0  /* We disable alpha blending here, as hid_expose_callback() does not
+        * call set_layer() as appropriate for us to sub-composite rendering
+        * from each layer when drawing a single element. If we leave alpha-
+        * blending on, it means text and overlapping pads are rendered ugly.
+        */
+
   glEnable (GL_BLEND);
   glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+#endif
 
   glViewport (0, 0, allocation.width, allocation.height);
